### PR TITLE
sci-libs/orocos_kdl: bugday 2021-04-03 related fixes

### PR DESCRIPTION
**Pull request by waebbl**
sci-libs/orocos_kdl: bump to EAPI 7

Port to cmake.eclass using ninja
Bug: https://bugs.gentoo.org/770595

drop 1.4.0

backport 1.4.0 changes to live ebuild
Backport changes from 1.4.0-r1 and update ebuild:
- bump to EAPI 7
- port to cmake.eclass using ninja
Bugday 2021-04-03

Closes: https://bugs.gentoo.org/770595
Package-Manager: Portage-3.0.18, Repoman-3.0.3
Signed-off-by: Bernd Waibel <waebbl-gentoo@posteo.net>

All 3 commits related to Bugday 2021-04-03

### CODE REVIEW

File: sci-libs/orocos_kdl/orocos_kdl-1.4.0-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	S="${WORKDIR}/${P}/${PN}"
+else
+	SRC_URI="https://github.com/orocos/orocos_kinematics_dynamics/archive/v${PV}.tar.gz -> orocos_kinematics_dynamics-${PV}.tar.gz"
+	KEYWORDS="~amd64 ~arm ~x86"
+	S="${WORKDIR}/orocos_kinematics_dynamics-${PV}/${PN}"
+fi
+
+LICENSE="LGPL-2.1"
+SLOT="0/14"
+IUSE="doc test examples models"
```
Comment (a17r) [inline sci-libs/orocos_kdl/orocos_kdl-1.4.0-r1.ebuild:23]: sort `IUSE`

File: sci-libs/orocos_kdl/orocos_kdl-1.4.0-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	cmake_src_compile
+	if use doc; then
+		cmake_src_compile docs
+		rm "${BUILD_DIR}/doc/kdl.tag" || die
+	fi
+}
+
+src_test() {
+	cd "${BUILD_DIR}" || die
+	eninja check
```
Comment (a17r) [inline sci-libs/orocos_kdl/orocos_kdl-1.4.0-r1.ebuild:65]:
```
pushd "${BUILD_DIR}" > /dev/null || die
    eninja check
popd > /dev/null || die
```

File: sci-libs/orocos_kdl/orocos_kdl-1.4.0-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	doc? ( app-doc/doxygen[dot] )
+	test? ( dev-util/cppunit )
+"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-0001-don-t-install-kdl.tag-file.patch
+	"${FILESDIR}"/${P}-0002-add-LIB_SUFFIX-in-pkg-config-file.patch
+)
+
+DOCS=( README )
```
Comment (a17r) [inline sci-libs/orocos_kdl/orocos_kdl-1.4.0-r1.ebuild:39]: You can drop that unless it would cause useless files to be installed

**Review by a17r — CHANGES_REQUESTED** [review]

**Comment by a17r** [comment]
Slightly weird wording - I was confused about "backporting" 1.4.0 changes and had to look at stats to find it was about the live ebuild (intuitively would rather be "forward" port although I would not use that wording).

**Comment by waebbl** [comment]
> Slightly weird wording - I was confused about "backporting" 1.4.0 changes and had to look at stats to find it was about the live ebuild (intuitively would rather be "forward" port although I would not use that wording).

yes it sound a bit weird due to the 'higher' version number and being the newer code base.

**Pull request closed without merging**